pages/edit.js

```diff
@@ -60,21 +76,29 @@ export default function Edit({ items, recipes, shoppingItems }) {
         <link rel="icon" href="/favicon.ico" />
       </Head>
 
-      <Header>Liste bearbeiten</Header>
+      {session ? (
+        <Header text="Liste bearbeiten">
+          <SignOutButton onSignOut={signOut} />
+        </Header>
+      ) : (
+        <Header text="MyShoppingManager" isOverlappingAnimation={true} />
+      )}
       <main>
         <Background opacity="0.7" />
-        <ContentWrapper>
-          <ShoppingListEditor
-            items={items}
-            recipes={recipes}
-            listItems={listItems}
-            onDelete={deleteItem}
-            onAdd={addItem}
-          />
-        </ContentWrapper>
+        {session && (
+          <ContentWrapper>
+            <ShoppingListEditor
+              items={items}
+              recipes={recipes}
+              listItems={listItems}
+              onDelete={deleteItem}
+              onAdd={addItem}
+            />
+          </ContentWrapper>
+        )}
+        {!session && <SignIn onSignIn={signIn} />}
```

Comment:
I guess this could be one expression, too.

```suggestion
        {session ? (
          <ContentWrapper>
            <ShoppingListEditor
              items={items}
              recipes={recipes}
              listItems={listItems}
              onDelete={deleteItem}
              onAdd={addItem}
            />
          </ContentWrapper>
        ) :
         <SignIn onSignIn={signIn} />}
```